packages/session-replay-browser/src/config/joined-config.ts

```diff
@@ -1,26 +1,24 @@
-import {
-  SessionReplaySessionIDBStore as AmplitudeSessionReplaySessionIDBStore,
-  SessionReplayOptions,
-} from '../typings/session-replay';
+import { createRemoteConfigFetch, RemoteConfigFetch } from '@amplitude/analytics-remote-config';
+import { SessionReplayOptions } from '../typings/session-replay';
 import { SessionReplayLocalConfig } from './local-config';
-import { SessionReplayRemoteConfigFetch } from './remote-config';
 import {
   SessionReplayLocalConfig as ISessionReplayLocalConfig,
-  SessionReplayRemoteConfigFetch as ISessionReplayRemoteConfigFetch,
   SessionReplayJoinedConfig,
+  SessionReplayRemoteConfig,
 } from './types';
 
 export class SessionReplayJoinedConfigGenerator {
   localConfig: ISessionReplayLocalConfig;
-  remoteConfigFetch: ISessionReplayRemoteConfigFetch;
-  sessionIDBStore: AmplitudeSessionReplaySessionIDBStore;
+  remoteConfigFetch: RemoteConfigFetch<SessionReplayRemoteConfig> | undefined;
 
-  constructor(apiKey: string, options: SessionReplayOptions, sessionIDBStore: AmplitudeSessionReplaySessionIDBStore) {
-    this.sessionIDBStore = sessionIDBStore;
+  constructor(apiKey: string, options: SessionReplayOptions) {
     this.localConfig = new SessionReplayLocalConfig(apiKey, options);
-    this.remoteConfigFetch = new SessionReplayRemoteConfigFetch({
+  }
+
+  async initialize() {
+    this.remoteConfigFetch = await createRemoteConfigFetch<SessionReplayRemoteConfig>({
       localConfig: this.localConfig,
-      sessionIDBStore: this.sessionIDBStore,
+      configKeys: ['sessionReplay'],
```

Comment:
Since config keys are shared across multiple SDKs and plugins I think it would be good to have a way to structure/track them. 

Before there was a ConfigNamespace in the Remote config package.
https://github.com/amplitude/Amplitude-TypeScript/pull/745/files#diff-b803bf7e9e25d71172f51d5b97ec5108f57dc7d115f57ac084e6cf3ad2504cb7L12-L14

I was thinking we could instead define a ConfigNamespace in the SDK/Plugin that uses it.

To enforce via the type system, what about making `ConfigNamspace` a tagging interface that should be implemented by users of the remote config? e.g.

~/remote-config/
```
export interface ConfigNamespace {
  key: string
}
```

~/session-replay-borwser/
```
export SessionReplayConfigNamespace extends ConfigNamespace {
  key: "sessionReplay"
}
```

Reply:
I just tried this in code, and it ends up reading pretty confusingly, as I end up needing to refer to `ConfigNamespace['key'] in all places I'm using the ConfigNamespace type, as I really only need the string value. Then I tried just doing `export type ConfigNamespace = string`, but this can't be extended, and you can't create an interface from a primitive like string. 

Maybe I'm just getting blind to the right solution here in typescript at the moment (I've been wrestling with it all day), but I think this might overcomplicate things without a lot of payoff. I will say this is why I liked the enum I had before - we could add an enum for ConfigNamespace somewhere generic like the `@amplitude/analytics-types` package?